lib/web/assistant.go

```diff
@@ -275,6 +283,9 @@ func runAssistant(h *Handler, w http.ResponseWriter, r *http.Request, sctx *Sess
 		// write user message to both in-memory chain and persistent storage
 		chat.Insert(openai.ChatMessageRoleUser, wsIncoming.Payload)
 
+		// TODO(justinas): should we handle this error?
+		promptTokens, _, _ := tokenizerInstance.Encode(wsIncoming.Payload)
```

Comment:
This is not accurate.  `Payload` contains only the string from a user. At the beginning of each conversation, we add out system prompt that is not included here https://github.com/gravitational/teleport/blob/9c405c7fd59bd1144a207538b3148e44d4208167/lib/ai/prompt.go#L26.
Then each query contains all previous messages and responses (to remember the context).

Reply:
Updated the calculation to take history into the account, made other adjustments to match the OpenAI calculations.